src/azul/service/manifest_service.py

```diff
@@ -405,34 +410,48 @@ def get_cached_manifest_with_object_key(self,
                                             object_key: str
                                             ) -> Optional[Manifest]:
         generator = ManifestGenerator.for_format(format_, self, catalog, filters)
-        presigned_url = self._get_cached_manifest(generator, object_key)
-        if presigned_url is None:
-            return None
-        else:
-            return Manifest(location=presigned_url,
-                            was_cached=True,
+        file_name, was_cached = self._get_cached_manifest_file_name(generator, object_key)
+        if was_cached:
+            return Manifest(location=self._presign_url(object_key, file_name),
+                            was_cached=was_cached,
                             format_=format_,
                             catalog=catalog,
                             filters=filters,
-                            object_key=object_key)
-
-    def _get_cached_manifest(self,
-                             generator: 'ManifestGenerator',
-                             object_key: str
-                             ) -> Optional[str]:
-        if self._can_use_cached_manifest(object_key):
-            file_name = self._use_cached_manifest(generator, object_key)
-            return self._presign_url(object_key, file_name)
+                            object_key=object_key,
+                            file_name=file_name)
         else:
             return None
 
     file_name_tag = 'azul_file_name'
 
-    def _use_cached_manifest(self, generator: 'ManifestGenerator', object_key: str) -> Optional[str]:
+    def _get_cached_manifest_file_name(self,
+                                       generator: 'ManifestGenerator',
+                                       object_key: str
+                                       ) -> Tuple[Optional[str], bool]:
         """
-        Return the content disposition file name of the exiting cached manifest.
+        Return the content disposition file name of the exiting cached manifest
+        if the manifest was previously created, still exists in the bucket and
+        won't be expiring soon.
+
+        :param generator: A generator for a specific manifest format
+        :param object_key: S3 object key (eg. 'manifests/e0fabf97-7abb-5111-af97-810f1e736c71.tsv'
         """
-        if generator.use_content_disposition_file_name:
+        try:
+            response = self.storage_service.head(object_key)
+        except self.storage_service.client.exceptions.ClientError as e:
+            if int(e.response['Error']['Code']) == 404:
+                logger.info('Cached manifest not found: %s', object_key)
+                use_cached_manifest = False
```

Comment:
You could just return here. Setting the flag prevents anything else happening in the method.